lib/server.ml

```diff
@@ -139,6 +139,15 @@ let connect channel ?offer () =
         else negotiate_tls make_tls_channel
       )
 
+let with_connection clearchan ?offer f =
+  (match offer with
+    | None -> connect clearchan ()
+    | Some offer -> connect clearchan ~offer ())
+  >>= fun (_name, t) ->
```

Comment:
We need to pass `_name` to `f`, because `xapi-nbd` will need to know the export name.